tests/unit/resources/test_compensation.py

```diff
@@ -98,3 +108,100 @@ def test_create_from_spill_string(spillstring):
         "Qdot655-A",
         "Qdot705-A",
     ]
+
+
+@responses.activate
+def test_apply_comp_errors_for_nonmatching_channels(
+    client, ENDPOINT_BASE, compensation, fcs_file
+):
+    events_body = open("tests/data/Acea - Novocyte.fcs", "rb")
+    responses.add(
+        responses.GET,
+        f"{ENDPOINT_BASE}/experiments/{EXP_ID}/fcsfiles/{fcs_file._id}.fcs",
+        body=events_body,
+    )
+    events = parse_fcs_file(client.download_fcs_file(EXP_ID, fcs_file._id))
+    fcs_file.events = events
+
+    with pytest.raises(IndexError):
+        compensation.apply(fcs_file)
+
+
+@responses.activate
+def test_apply_compensation_to_fcs_file_with_no_events(
```

Comment:
I think this is the test I suggested removing. The test title here and comment on 134 don't match the actual test behavior. Should be adequately covered by the new test at the end.

130-151 ✂️